src/AuthContext.tsx

```diff
@@ -80,8 +80,7 @@ export const AuthProvider: FC<AuthProviderProps> = ({
   ...props
 }) => {
   const [userData, setUserData] = useState<User | null>(null);
-
-  const userManager = initUserManager(props);
+  const [userManager] = useState<UserManager>(initUserManager(props));
```

Comment:
I'm unsure about this. Isn't it better using `useEffect` for `initUserManager` as well? 🤷

Reply:
Good point, if the props would change, the `useEffect` would help.

ala
```
useEffect(() => {
    setUserManager(initUserManager(props));
  }, [props]);
```

On the other hand: In which application changes the config props for the authentication at runtime? None. Lets keep this simple.
BTW the auth0-react library is doing it similarly:
https://github.com/auth0/auth0-react/blob/feaf7fd10ce19c3c2cbab30c57c5bfdb07c5ab40/src/auth0-provider.tsx#L221

Important here is that the userManager is in a state and has the type `UserManager` (not null or undefined).

What is strange is that `initUserManager` is an exported function. I can fix this by an additional merge request.

Reply:
please merge this as it